hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/TopologySubcommand.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hdds.scm.cli;
+
+import org.apache.hadoop.hdds.cli.HddsVersionProvider;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos;
+import org.apache.hadoop.hdds.scm.client.ScmClient;
+import picocli.CommandLine;
+import static org.apache.hadoop.hdds.protocol.proto.HddsProtos.NodeState.DEAD;
+import static org.apache.hadoop.hdds.protocol.proto.HddsProtos.NodeState.DECOMMISSIONED;
+import static org.apache.hadoop.hdds.protocol.proto.HddsProtos.NodeState.DECOMMISSIONING;
+import static org.apache.hadoop.hdds.protocol.proto.HddsProtos.NodeState.HEALTHY;
+import static org.apache.hadoop.hdds.protocol.proto.HddsProtos.NodeState.STALE;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+/**
+ * Handler of printTopology command.
+ */
+@CommandLine.Command(
+    name = "printTopology",
+    description = "Print a tree of the network topology as reported by SCM",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class)
+public class TopologySubcommand implements Callable<Void> {
+
+  @CommandLine.ParentCommand
+  private SCMCLI parent;
+
+  private static List<HddsProtos.NodeState> stateArray = new ArrayList<>();
+
+  static {
+    stateArray.add(HEALTHY);
+    stateArray.add(STALE);
+    stateArray.add(DEAD);
+    stateArray.add(DECOMMISSIONING);
+    stateArray.add(DECOMMISSIONED);
+  }
+
+  @Override
+  public Void call() throws Exception {
+    try (ScmClient scmClient = parent.createScmClient()) {
+      for (HddsProtos.NodeState state : stateArray) {
+        List<HddsProtos.Node> nodes = scmClient.queryNode(state,
+            HddsProtos.QueryScope.CLUSTER, "");
+        if (nodes != null && nodes.size() > 0) {
+          // show node state
+          System.out.println("State = " + state.toString());
```

Comment:
Thanks @ChenSammi  for the patch. It LGTM, +1 pending CI. 
One question: should we have the option to organize the node based on topology in addition to the state?

Reply:
@xiaoyuyao , we can add option to order the output acccording to topology layer. For example, for /rack/node topolgy, we can show, 
State = HEALTHY
/default-rack: 
ozone_datanode_1.ozone_default/172.18.0.3   
ozone_datanode_2.ozone_default/172.18.0.2   
ozone_datanode_3.ozone_default/172.18.0.4   
/rack1:
ozone_datanode_4.ozone_default/172.18.0.5   
ozone_datanode_5.ozone_default/172.18.0.6   
For /dc/rack/node topology, we can either show 
State = HEALTHY
/default-dc/default-rack: 
ozone_datanode_1.ozone_default/172.18.0.3   
ozone_datanode_2.ozone_default/172.18.0.2   
ozone_datanode_3.ozone_default/172.18.0.4   
/dc1/rack1:
ozone_datanode_4.ozone_default/172.18.0.5   
ozone_datanode_5.ozone_default/172.18.0.6   

or 

State = HEALTHY
default-dc: 
  default-rack: 
ozone_datanode_1.ozone_default/172.18.0.3   
ozone_datanode_2.ozone_default/172.18.0.2   
ozone_datanode_3.ozone_default/172.18.0.4   
dc1:
  rack1:
ozone_datanode_4.ozone_default/172.18.0.5   
ozone_datanode_5.ozone_default/172.18.0.6   

This can be done in a follow on JIRA. Currently, this plain format just meets our needs.

Reply:
Agree, let's add that in follow up JIRA. I will merge it shortly.